backend/dataall/core/environment/services/managed_iam_policies.py

```diff
@@ -29,42 +29,69 @@ def policy_type(self):
+        Returns string of policy name with index at the end .Needs to be implemented in the ManagedPolicies inherited classes
         """
         ...
 
     @abstractmethod
-    def create_managed_policy_from_inline_and_delete_inline(self) -> str:
+    def generate_empty_policy(self) -> dict:
         """
-        Returns policy arn and needs to be implemented in the ManagedPolicies inherited classes
-        It is used for backwards compatibility. It should be deprecated and removed in future releases.
+        Returns dict and needs to be implemented in the ManagedPolicies inherited classes
         """
         ...
 
-    def check_if_policy_exists(self) -> bool:
-        policy_name = self.generate_policy_name()
+    def check_if_policy_exists(self, policy_name) -> bool:
         share_policy = IAM.get_managed_policy_by_name(self.account, self.region, policy_name)
         return share_policy is not None
 
-    def check_if_policy_attached(self):
-        policy_name = self.generate_policy_name()
-        return IAM.is_policy_attached(self.account, self.region, policy_name, self.role_name)
-
-    def attach_policy(self):
-        policy_arn = f'arn:aws:iam::{self.account}:policy/{self.generate_policy_name()}'
-        try:
-            IAM.attach_role_policy(self.account, self.region, self.role_name, policy_arn)
-        except Exception as e:
-            raise Exception(f"Required customer managed policy {policy_arn} can't be attached: {e}")
+    def get_managed_policies(self) -> List[str]:
+        policy_pattern = self.generate_base_policy_name()
+        share_policies = self._get_policy_names(policy_pattern)
+        return share_policies
+
+    def check_if_policy_attached(self, policy_name):
+        is_policy_attached = IAM.is_policy_attached(self.account, self.region, policy_name, self.role_name)
+        return is_policy_attached
+
+    def get_policies_unattached_to_role(self):
+        policy_pattern = self.generate_base_policy_name()
+        share_policies = self._get_policy_names(policy_pattern)
+        unattached_policies = []
+        for share_policy_name in share_policies:
+            if not self.check_if_policy_attached(share_policy_name):
+                unattached_policies.append(share_policy_name)
+        return unattached_policies
+
+    def attach_policies(self, managed_policies_list: List[str]):
+        for policy_name in managed_policies_list:
+            policy_arn = f'arn:aws:iam::{self.account}:policy/{policy_name}'
+            try:
+                IAM.attach_role_policy(self.account, self.region, self.role_name, policy_arn)
+            except Exception as e:
+                raise Exception(f"Required customer managed policy {policy_arn} can't be attached: {e}")
+
+    def _get_policy_names(self, policy_pattern):
+        share_policies = IAM.list_policy_names_by_policy_pattern(self.account, self.region, policy_pattern)
```

Comment:
Remove specific references to share_managed_policies. Instead of `share_policies` --> `managed_policies`